src/services/cipher.service.ts

```diff
@@ -295,9 +295,10 @@ export class CipherService implements CipherServiceAbstraction {
     async getAllDecrypted(): Promise<CipherView[]> {
         if (this.decryptedCipherCache != null) {
             const userId = await this.userService.getUserId();
-            if ((this.searchService().indexedEntityId ?? userId) !== userId)
+            if (this.searchService().indexedEntityId ?? userId !== userId)
             {
-                await this.searchService().indexCiphers();
+                await this.searchService().awaitAndClearIndex();
+                this.searchService().indexCiphers();
```

Comment:
Wouldn't this also need to be `await`'ed since `.indexCiphers()` still returns a `Promise<void>`?

Reply:
yes, thank you

Reply:
awaiting this method caused things to break again, pointing me off in a new direction again. The flow of this code is really intermingled. ☹️